### ISSUE TITLE
`use-isnan` fails when use NaN in `||` operator

### ISSUE THREAD

**Issue by timocov** [palantir/tslint#2255]
### Bug Report

- __TSLint version__: 4.3.0
- __TypeScript version__: 2.2.1
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
declare var a: number | undefined;
console.log(a || NaN);
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "use-isnan": true
  }
}
```

#### Actual behavior
`Found an invalid comparison for NaN: a || NaN`

#### Expected behavior
No errors

**Comment by ajafff** [comment]
But that's not a comparison.
Your code does basically the following: If a is truthy (`a !== 0 && a !== undefined && !isNan(a)`) log a, else log NaN.
There's nothing wrong in this piece of code and nothing the `use-isnan`-Rule could detect.

**Comment by sbj42** [comment]
I believe @timocov would agree with you, that `use-isnan` should not complain about that code, but it does. I was able to reproduce it by adding the following lines to `rules/use-isnan/test.ts.lint`:

```ts
// no violation for logical operators
x = 0 || NaN;
x = 0 && NaN;
```

Then ran the tests and got:

```
// no violation for logical operators
x = 0 || NaN;
    ~~~~~~~~  [Found an invalid comparison for NaN: 0 || NaN]
x = 0 && NaN;
    ~~~~~~~~  [Found an invalid comparison for NaN: 0 && NaN]
```

The implementation of `use-isnan` has an exception only for the simple assignment operator, but TypeScript has many other operators which are not "comparison" operators.  Many of them would be very silly to use with NaN, but nevertheless the rule does not correctly limit its scope.

**Comment by ajafff** [comment]
Ah, sorry. I misread the opening post.

@sbj42 you already analyzed the problem. Fixing this should not be that difficult. Would you like to fix it or should I go for it?

**Comment by sbj42** [comment]
I'd like to fix it.  I'll give it a try later today.

**Comment by sbj42** [comment]
I believe this is now resolved.
